### ISSUE TITLE
Save list of visible panes before closing Spyder and restore them on the next session

### ISSUE THREAD

**Issue by FObersteiner** [spyder-ide/spyder#12104]
### Problem Description
* When you open Spyder, IPython console, Editor and the Help pane are shown by default.
![Anmerkung 2020-04-02 103321](https://user-images.githubusercontent.com/31826595/78227388-78154900-74cd-11ea-8cb0-62c6ba508f84.png)

* **--> however, I'd like to have another pane to be selected at startup**, e.g. "_Variable Explorer_" (or whatever I had selected when I closed Spyder the last time)

* To my knowledge (info from @ccordoba12), at the moment there is no way to change this behavior e.g. in the settings. So this is basically a suggestion for a new feature.

### What steps reproduce the problem?
* open Spyder

### What is the expected output? What do you see instead?
(not applicable)

### Paste Traceback/Error Below (if applicable)
(not applicable)

## Versions
* Spyder version: 4.1.1
* Python version: 3.8.2
* Qt version: 5.12.3
* PyQt version: 1.9.0
* Operating System name/version: Windows 10 (1909)

### Dependencies
(not applicable)

**Comment by goanpeca** [comment]
Hi @MrFuppes yes, this is something I also would like. Will work on it some time in April,

**Comment by royalbisht** [comment]
This problem is back after 5.0 to 5.0.3. Please help.
![Screenshot (10)](https://user-images.githubusercontent.com/16011976/118924258-753c1880-b95a-11eb-8444-214f5d55b86c.png)

**Comment by MrFuppes** [comment]
@royalbisht I don't think this feature was implemented yet. Not sure where it is on the roadmap ;-)

**Comment by ilayn** [comment]
I have been also experiencing this. Not only the help pane is switched by default, when all panes except the console and editor  are closed, upon startup it enables itself by disrupting 2-pane the layout. Hence after every restart, it causes to be re-closed and the remaining panes to be realigned. Saving layout, locking unlocking panes all combinations lead to no avail.

**Comment by abudden** [comment]
This is the one thing that I find to be most frustrating with Spyder (although to be fair it's a minor irritation, which says a lot about how good Spyder is).  Every time I start Spyder I have to click on "Variable Explorer".  I have literally never used the "Help" pane.

I closed the "Plot" pane (as I find it causes more problems than it solves) and it has not re-opened when I restart Spyder.  If I close the "Help" pane and restart Spyder, the help pane re-opens and is focused again.  The Variable Explorer is **so** much more useful.

Please allow the Variable Explorer to be made the default (or just remember the pane that was focused when exiting, or remember that the help pane was closed).

**Comment by abudden** [comment]
For anyone else sufficiently frustrated with this to hack the source code, commenting out these lines of code seem to fix the problem:

```patch
diff -ur old/app/mainwindow.py new/app/mainwindow.py
--- old/app/mainwindow.py
+++ new/app/mainwindow.py
@@ -1199,8 +1199,8 @@
 
         # Show Help and Consoles by default
         plugins_to_show = [self.ipyconsole]
-        if self.help is not None:
-            plugins_to_show.append(self.help)
+        #if self.help is not None:
+        #    plugins_to_show.append(self.help)
         for plugin in plugins_to_show:
             if plugin.dockwidget.isVisible():
                 plugin.dockwidget.raise_()
diff -ur old/plugins/help/plugin.py new/plugins/help/plugin.py
--- old/plugins/help/plugin.py
+++ new/plugins/help/plugin.py
@@ -238,7 +238,7 @@
 
     def show_intro_message(self):
         """Show the IPython introduction message."""
-        self.switch_to_plugin()
+        #self.switch_to_plugin()
         self.get_widget().show_intro_message()
 
     def show_rich_text(self, text, collapse=False, img_path=''):

```

**Comment by MrFuppes** [comment]
actually, Spyder 5.12.3 works just as expected for me (pip install, on Windows 10). I was so getting used to this annoying auto-select of the help pane that I didn't even realized the "feature" was removed... 

cheers to the devs! ☕

**Comment by abudden** [comment]
> actually, Spyder 5.12.3 works just as expected for me (pip install, on Windows 10). I was so getting used to this annoying auto-select of the help pane that I didn't even realized the "feature" was removed...
> 
> cheers to the devs! ☕

That's great to hear.  I'm using conda-forge and the latest version is 5.1.5, which needed the patch I posted.  I'm reasonably happy with the patch, but I'm glad that when I next upgrade it'll hopefully not need patching again.

**Comment by abudden** [comment]
Looks like it was fixed as a result of #15331 and #17222

**Comment by ccordoba12** [comment]
> This is the one thing that I find to be most frustrating with Spyder (although to be fair it's a minor irritation, which says a lot about how good Spyder is)

Thanks for your kind works @abudden!

> Looks like it was fixed as a result of https://github.com/spyder-ide/spyder/issues/15331 and https://github.com/spyder-ide/spyder/pull/17222, released as part of 5.2.2. Hopefully that will make it to conda-forge soon

It's already in conda-forge, so you can update anytime you want. It's not available in Anaconda yet.

> or just remember the pane that was focused when exiting

Yeah, that's what's really missing and I'll try to implement it in a future release to finally close this issue.
